rotkehlchen/db/history_events.py

```diff
@@ -608,10 +620,17 @@ def get_history_events_count(
         count_without_limit = cursor.execute(query, bindings).fetchone()[0]
 
         if entries_limit is not None:
+            free_query_group_by = maybe_filter_ignore_asset(query_filter)
+            prepared_query, bindings = query_filter.prepare(
+                with_group_by=False,
+                special_free_query=True,
+                with_order=False,
+                with_pagination=False,
+            )
+
             query = 'SELECT * ' + query_filter.get_join_query()
-            if group_by_event_ids:  # we take the groups before the limit has been applied
-                query += ' GROUP BY event_identifier '
-            query += ' ORDER BY timestamp DESC LIMIT ?'
+            # we take the groups before the limit has been applied
+            query += free_query_group_by + 'GROUP BY event_identifier ORDER BY timestamp DESC, sequence_index ASC LIMIT ?'  # noqa: E501
```

Comment:
same, don't do it in two operations if it fits in one

```suggestion
            # we take the groups before the limit has been applied
            query = (
                f'SELECT * {query_filter.get_join_query()} {free_query_group_by}'
                'GROUP BY event_identifier ORDER BY timestamp DESC, sequence_index ASC LIMIT ?'
            )
```